src/hotspot/share/gc/parallel/psScavenge.cpp

```diff
@@ -180,55 +180,64 @@ class PSKeepAliveClosure: public OopClosure {
 class PSEvacuateFollowersClosure: public VoidClosure {
  private:
   PSPromotionManager* _promotion_manager;
+  TaskTerminator* _maybe_terminator;
```

Comment:
What does the `_maybe` in `_maybe_terminator` indicate? Just naming it `_terminator` seems as fine as a `nullptr` seems like a valid value.
It is explained in `PSRefProcClosureContext`, but it seems too much to add this everywhere where a pointer could be a valid `nullptr`.

Reply:
Maybe indicates (Just a valid pointer) or (Nothing nullptr), optional might be a better prefix? 

I would agree to remove the prefix if we would consistently use references where nullptr is not a valid state. If we are going to use pointers instead of references because it visualizes that we take the address at call site (something I disagree with), I think adding `maybe` or `optional` is a (visual) safeguard for not de-referencing the pointer without verifying it can safely be de-referenced.

I will remove the prefix if you do not buy my explanation.